tenpy/simulations/time_evolution.py

```diff
@@ -74,3 +80,347 @@ def final_measurements(self):
+            raise TypeError("Ground state must be an MPS")
+
+        data_options = gs_data['simulation_parameters']
+        for key in data_options.keys():
+            if not isinstance(key, str) or not key.startswith('model'):
+                continue
+            if key not in self.options:
+                self.options[key] = data_options[key]
+            elif self.options[key] != data_options[key]:
+                raise ValueError("Different model parameters in GroundStateSearch and GroundStateSearch")
+
+        if 'energy' in gs_data.keys():
+            self.gs_energy = self.options['gs_energy'] = gs_data['energy']
+
+    def _get_operator_t0(self):
+        """Converts the specified operators and indices into a list of tuples [(op1, i_1), (op2, i_2)]"""
+        idx = self.operator_t0_config.get('i', self.psi.L // 2)
+        ops = self.operator_t0_config.get('op', 'Sigmay')
+        ops = [ops] if not isinstance(ops, list) else ops  # pass ops as list
+        form = self.operator_t0_config.get('idx_form', 'mps')
+        if form not in ['mps', 'lat']:
+            raise ValueError("the idx_form must be either mps or lat")
+        if form == 'mps':
+            idx = list(idx if isinstance(idx, list) else [idx])
+        else:
+            if not isinstance(idx, list):
+                raise TypeError("for idx_form lat, i must be given as list [x, y, u] or list of lists")
+            mps_idx = self.model.lat.lat2mps_idx(idx)
+            idx = list(mps_idx) if isinstance(mps_idx, np.ndarray) else [mps_idx]  # convert to mps index
+
+        if len(ops) > len(idx):
+            if len(idx) != 1:
+                raise ValueError("Ill-defined tiling: ops is longer than idx, and idx is not one")
+            idx = idx*len(ops)
+        elif len(ops) < len(idx):
+            if len(ops) != 1:
+                raise ValueError("Ill-defined tiling: idx is longer than ops, and ops is not one")
+            ops = ops * len(idx)
+        # generate list of tuples of form [(op1, i_1), (op2, i_2), ...]
+        op_list = list(zip(ops, idx))
+        return op_list
+
+    def apply_operator_t0_to_psi(self):
+        # TODO: think about segment boundary conditions
+        # TODO: make JW string consistent, watch for changes in apply_local_op to have autoJW
+        self.operator_t0 = self._get_operator_t0()
+        operator_t0 = self.operator_t0
+        if len(operator_t0) == 1:
+            op, i = operator_t0[0]
+            if self.model.lat.site(i).op_needs_JW(op):
+                for j in range(i):
+                    self.psi.apply_local_op(j, 'JW')
+            self.psi.apply_local_op(i, op)  # TODO: check if renormalize=True makes sense here
+        else:
+            ops, i_min, _ = self.psi._term_to_ops_list(operator_t0, True)  # applies JW string automatically
+            for i, op in enumerate(ops):
+                self.psi.apply_local_op(i_min + i, op)
+
+    def prepare_results_for_save(self):
```

Comment:
Nice function! You can move this code directly into the base `Simulation` class.